### PR TITLE
Access to modification group form from sld

### PR DESCRIPTION
**Pull request by Walid-Sahnoun**

### CODE REVIEW

File: src/components/menus/base-equipment-menu.js [hunk 1]
```diff
@@ -83,15 +85,42 @@ const BaseEquipmentMenu = ({
...
+                                    className={classes.listItemText}
+                                    primary={
+                                        <Typography noWrap>
+                                            {intl.formatMessage({
+                                                id: 'edit',
+                                            })}
+                                        </Typography>
+                                    }
+                                ></ListItemText>
+                            </MenuItem>
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/components/menus/base-equipment-menu.js]: done

File: src/components/diagrams/diagram.js [hunk 2]
```diff
@@ -138,6 +140,20 @@ const Diagram = forwardRef((props, ref) => {
 
     const [errorMessage, setErrorMessage] = useState('');
 
+    const [equipmentToModify, setEquipmentToModify] = useState({
+        equipmentId: null,
+        equipmentType: null,
+    });
+
+    const openModificationDialog = (equipmentToModify) => {
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/components/diagrams/diagram.js]: renamed

File: src/components/diagrams/diagram.js [hunk 3]
```diff
@@ -138,6 +140,20 @@ const Diagram = forwardRef((props, ref) => {
 
     const [errorMessage, setErrorMessage] = useState('');
 
+    const [equipmentToModify, setEquipmentToModify] = useState({
+        equipmentId: null,
+        equipmentType: null,
+    });
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/components/diagrams/diagram.js]: done

File: src/components/diagrams/diagram.js [hunk 4]
```diff
@@ -138,6 +140,20 @@ const Diagram = forwardRef((props, ref) => {
...
+
+    const openModificationDialog = (equipmentToModify) => {
+        closeEquipmentMenu();
+        setEquipmentToModify(equipmentToModify);
+    };
+
+    const closeModificationDialog = () => {
+        setEquipmentToModify({ equipmentId: null, equipmentType: null });
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/components/diagrams/diagram.js]: done

File: src/components/diagrams/diagram.js [hunk 5]
```diff
@@ -138,6 +140,20 @@ const Diagram = forwardRef((props, ref) => {
...
 
+    const [equipmentToModify, setEquipmentToModify] = useState({
+        equipmentId: null,
+        equipmentType: null,
+    });
+
+    const openModificationDialog = (equipmentToModify) => {
+        closeEquipmentMenu();
+        setEquipmentToModify(equipmentToModify);
+    };
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/components/diagrams/diagram.js]: done

File: src/components/menus/equipment-menu.js [hunk 6]
```diff
@@ -19,7 +19,13 @@ const useStyles = makeStyles((theme) => ({
 
 const withEquipmentMenu =
     (BaseMenu, menuId, equipmentType) =>
-    ({ id, position, handleClose, handleViewInSpreadsheet }) => {
+    ({
+        id,
+        position,
+        handleClose,
+        handleViewInSpreadsheet,
+        openModificationDialog,
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/components/menus/equipment-menu.js]: done

**Review by niconoir — APPROVED** [review]

File: src/components/menus/base-equipment-menu.js [hunk 7]
```diff
@@ -83,15 +109,29 @@ const BaseEquipmentMenu = ({
...
+                        <ItemViewInSpreadsheet
+                            key="ViewOnSpreadsheet"
+                            equipmentType={equipmentType}
+                            equipmentId={equipmentId}
+                            itemText={intl.formatMessage({
+                                id: 'ViewOnSpreadsheet',
+                            })}
+                            handleViewInSpreadsheet={handleViewInSpreadsheet}
+                        />
+                        {equipmentType === equipments.generators && (
```
Comment (Meklo) [inline src/components/menus/base-equipment-menu.js:122]: Why is it nested under `{equipmentType !== equipments.substations && equipmentType !== equipments.voltageLevels` ?
Couldn't it be on its own ?

File: src/components/dialogs/generator-modification-dialog.js [hunk 8]
```diff
@@ -198,12 +203,13 @@ const GeneratorModificationDialog = ({
     }, [generatorId, formValueEquipmentId]);
 
     useEffect(() => {
-        if (id) {
+        if (id || defaultIdValue) {
+            let key = id !== '' ? id : defaultIdValue;
```
Comment (Meklo) [inline src/components/dialogs/generator-modification-dialog.js:207]: Maybe you could replace it by `let key = id ? id : defaultIdValue;` ?

**Review by EstherDarkish — APPROVED** [review]
Code OK, test OK

File: src/components/menus/base-equipment-menu.js (hunk 7 above)
Comment (Walid-Sahnoun) [reply src/components/menus/base-equipment-menu.js]: yes, i move it for it's own bloc

**Review by Meklo — APPROVED** [review]

**Pull request merged**